protected/views/search/_new_result.php

```diff
@@ -1,73 +1,102 @@
-                <div class="span1" style="margin-left: 40px;margin-top: 10px;height: 30px;width: 40px"><div class="text-icon text-icon-sm text-icon-green" style="margin-right: 0px;">S
-            </div></div>
-                <div class="span8 main-content" style="float:right">
-                    <ul class="nav nav-tabs nav-stacked result-cell">
-                      <li><a class="result-main-link" href="<?= $dt['shorturl'] ?>"><?php echo $sample['name'] ?></a></li>
-                      <li>
-                        <strong>
-                            <?= $sample['species_common_name'] ?>
-                            NCBI taxonomy : 
-                            <a class="result-sub-links" target="_blank" href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=<?php echo $sample['species_tax_id'] ?>">
-                                <?= $sample['species_tax_id'] ?>
-                            </a>
-                        </strong>
-                      </li>
-                      <li class="searchID"><a class="result-sub-links" href="<?php echo $dt['shorturl'] ?>">DOI:10.5524/<?= $dt['identifier'] ?></a></li>
-                    </ul>
+                                    $listItems = '';
+                                    $numberOfLinks = count($authorLinks);
+                                    foreach ($authorLinks as $index => $link) {
+                                        $listItems .= '<li>' . $link;
+                                        if ($index !== $numberOfLinks - 1) {
+                                            $listItems .= ';&nbsp;';
+                                        }
+                                        $listItems .= '</li>';
+                                    }
+
+                                    $unorderedList = '<ul class="search-result-list search-result-author-list">' . $listItems . '</ul>';
+
+                                    echo $unorderedList;
+                                ?>
+                            </div>
+                        <?php endif; ?>
+                        <div><?= Yii::t('app', 'DOI') ?>:<?php echo "10.5524/" . $dt['identifier'] ?></div>
+                    </div>
                 </div>
+
+                <?php if (in_array('sample', $display)) { ?>
+                    <ul class="search-result-list">
+                        <?php foreach ($dsamples as $sample) { ?>
+                            <!--Sample section-->
+                            <li class="search-result-row">
+                                <div class="search-result-icon-container">
+                                    <div class="text-icon text-icon-sm text-icon-green search-result-icon" aria-hidden="true">S</div>
+                                    <span class="sr-only">Sample</span>
+                                </div>
+                                <div class="search-result-content">
+                                    <h3 class="search-result-title">
```

Comment:
judging from the tests and the php conditionals, it looks like an item will always have one dataset. Since the dataset has one h3, the sample titles might alternatively be a h4. It depends on what is the preferred html structure for these items. I left it as h3 as from the layout it looks like the dataset and samples are at the same hierachy level

Reply:
From an editorial perspective, dataset is of higher hierarchy than samples and files (you can see that in display in the dataset page), but in the search function, I think the original intent was to give any searchable objects the same importance, as most scientists using our tool may just be interested on samples, or files which are really the important things more so than the dataset which is just and editorial artefact or a summary of what to expect in samples and files.

So I think you did the right thing here by keeping the original `h3` elements for all of them.
(the fact that you can filter out the dataset results goes along that line of thinking too).